app/models/order_cycle.rb

```diff
@@ -10,11 +10,11 @@ class OrderCycle < ActiveRecord::Base
 
   # These scope names are prepended with "cached_" because there are existing accessor methods
   # :incoming_exchanges and :outgoing_exchanges.
-  has_many :cached_incoming_exchanges, conditions: { incoming: true }, class_name: "Exchange"
-  has_many :cached_outgoing_exchanges, conditions: { incoming: false }, class_name: "Exchange"
+  has_many :cached_incoming_exchanges, -> { where incoming: true }, class_name: "Exchange"
+  has_many :cached_outgoing_exchanges, -> { where incoming: false }, class_name: "Exchange"
 
-  has_many :suppliers, source: :sender, through: :cached_incoming_exchanges, uniq: true
-  has_many :distributors, source: :receiver, through: :cached_outgoing_exchanges, uniq: true
+  has_many :suppliers, -> { uniq }, source: :sender, through: :cached_incoming_exchanges
+  has_many :distributors, -> { uniq }, source: :receiver, through: :cached_outgoing_exchanges
```

Comment:
this can be done in master, right?

Reply:
We can try it :+1:

Reply:
Nope, turn out this one can't be done before Rails 4.